crd-generator/cli/src/main/java/io/fabric8/crd/generator/cli/CRDGeneratorCLI.java

```diff
@@ -0,0 +1,317 @@
+  )
+  // spotless:on
+  void setParameters(List<SourceParameter> parameters) {
+    setCustomResourceClassNames(parameters);
+    setFilesToScan(parameters);
+  }
+
+  @Override
+  public void run() {
+    LoggingConfiguration.configureLogger(verbose);
+    List<String> allClasspathElements = getClasspathElements();
+
+    log.trace("Custom Resource Class Names: {}", customResourceClassNames);
+    log.trace("Files to scan: {}", filesToScan);
+    log.trace("Classpath: {}", allClasspathElements);
+
+    CustomResourceCollector customResourceCollector = new CustomResourceCollector()
+        .withClasspathElements(allClasspathElements)
+        .withFilesToScan(filesToScan)
+        .withForceIndex(forceIndex)
+        .withForceScan(forceScan)
+        .withCustomResourceClasses(customResourceClassNames)
+        .withIncludePackages(includedPackages)
+        .withExcludePackages(excludedPackages);
+
+    List<Class<? extends HasMetadata>> customResourceClasses = customResourceCollector.findCustomResourceClasses();
+
+    if (customResourceClasses.isEmpty()) {
+      throw new CustomResourceClassNotFoundException();
+    }
+
+    log.debug("Generating CRDs for {} Custom Resource classes", customResourceClasses.size());
+
+    File sanitizedOutputDirectory;
+    try {
+      sanitizedOutputDirectory = outputDirectory.getCanonicalFile();
+    } catch (IOException e) {
+      throw new CRDGeneratorCliException("Could not get canonical file for " + outputDirectory, e);
+    }
+
+    try {
+      Files.createDirectories(sanitizedOutputDirectory.toPath());
+    } catch (IOException e) {
+      throw new CRDGeneratorCliException(
+          "Could not create output directory at " + sanitizedOutputDirectory, e);
+    }
+
+    CRDGenerator crdGenerator = new CRDGenerator()
+        .customResourceClasses(customResourceClasses)
+        .withParallelGenerationEnabled(!parallelDisabled)
+        .withImplicitPreserveUnknownFields(implicitPreserveUnknownFields)
+        .inOutputDir(sanitizedOutputDirectory);
+
+    crdGenerationInfo = crdGenerator.detailedGenerate();
+    crdGenerationInfo.getCRDDetailsPerNameAndVersion().forEach((crdName, versionToInfo) -> {
+      getOut().printf("Generated CRD %s:%n", crdName);
+      versionToInfo.forEach(
+          (version, info) -> getOut().printf(" %s -> %s%n", version, info.getFilePath()));
+    });
+  }
+
+  CRDGenerationInfo getCrdGenerationInfo() {
+    return crdGenerationInfo;
+  }
+
+  String getDiagText() {
+    StringBuilder sb = new StringBuilder();
+    sb.append("\n");
+
+    if (!customResourceClassNames.isEmpty()) {
+      sb.append("Custom Resource class names:\n");
+      customResourceClassNames.forEach(fqcn -> sb.append(" ").append(fqcn).append("\n"));
+    }
+    if (filesToScan.isEmpty()) {
+      sb.append("Scan Paths: []\n");
+    } else {
+      sb.append("Scan Paths:\n");
+      filesToScan.forEach(f -> sb.append(" ").append(f.getPath()).append("\n"));
```

Comment:
Not 100% sure whether it's applicable but I see you had used `%n` in above method. Should this be platform dependent as well (i.e `System.lineSeparator()`) ?

Reply:
You are right, this must be platform independent. Will fix it.